src/preprocessing/point_in_poly/winding_number_horman.jl

```diff
@@ -0,0 +1,77 @@
+"""
+    WindingNumberHorman()
+
+Algorithm for inside-outside segmentation of a complex shape proposed by Horman et al. (2001).
+It is only supported for 2D shapes.
+[`WindingNumberHorman`](@rerf) might handle edge cases a bit better, since the winding number is an integer value.
+Also, it is faster than [`WindingNumberJacobson`](@ref) for 2D shapes with about more than 100 edges.
+"""
+struct WindingNumberHorman end
+
+function (point_in_poly::WindingNumberHorman)(shape, points; store_winding_number=false)
+    (; edge_vertices) = shape
+
+    inpoly = falses(size(points, 2))
+
+    winding_numbers = Float64[]
+    store_winding_number && (winding_numbers = resize!(winding_numbers, length(inpoly)))
+
+    h_unit = SVector(1.0, 0.0)
+    v_unit = SVector(0.0, 1.0)
+
+    function quadrant(dot_v, dot_h)
+        (dot_v >= 0.0 && dot_h > 0.0) && return 0
+        (dot_v > 0.0 && dot_h <= 0.0) && return 1
+        (dot_v <= 0.0 && dot_h < 0.0) && return 2
+        (dot_v < 0.0 && dot_h >= 0.0) && return 3
+    end
+
+    Threads.@threads for query_point in axes(points, 2)
```

Comment:
```suggestion
    @threaded points for query_point in axes(points, 2)
```
You can also use an array when no system is available. This will run `Polyester.@batch` when `points` is an `Array` and GPU code when it's a GPU array type.

Reply:
Ah thanks!